From 4ab4b239115d3e8f16a362ee19744fd45f040196 Mon Sep 17 00:00:00 2001
From: Shadi Ammouri <shadi@marvell.com>
Date: Mon, 28 Aug 2017 15:27:01 +0300
Subject: [PATCH 6/6] dts: musdk: a7k/8/8kp agnic

- The platform entry will be used by the agnic netdev driver.
- The uio entry will be used by the generic platform uio driver
  for exposing:
	1) agnic configuration registers to user-space
	2) msi registers

Change-Id: Ifb6a231e7c4cfb4c2f8d604b7b0709b960151992
Signed-off-by: Shadi Ammouri <shadi@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/43483
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
---
 .../bindings/net/marvell-armada-giu-nic-uio.txt    | 26 ++++++++++++++++++++++
 .../bindings/net/marvell-armada-giu-nic.txt        | 20 +++++++++++++++++
 arch/arm64/boot/dts/marvell/armada-ap806.dtsi      | 16 +++++++++++++
 arch/arm64/boot/dts/marvell/armada-ap810.dtsi      | 15 +++++++++++++
 4 files changed, 77 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/net/marvell-armada-giu-nic-uio.txt
 create mode 100644 Documentation/devicetree/bindings/net/marvell-armada-giu-nic.txt

diff --git a/Documentation/devicetree/bindings/net/marvell-armada-giu-nic-uio.txt b/Documentation/devicetree/bindings/net/marvell-armada-giu-nic-uio.txt
new file mode 100644
index 0000000..a9def97
--- /dev/null
+++ b/Documentation/devicetree/bindings/net/marvell-armada-giu-nic-uio.txt
@@ -0,0 +1,26 @@
+* Marvell Armada GIU NIC UIO
+
+This entry is used by the generic platform uio driver (uio_pdrv_genirq) for:
+1) mapping the gnic configuration registers to user-space.
+2) mapping the msi registers to user-space.
+
+required properties:
+=========================
+- compatible: should be "marvell,armada-giu-nic-uio"
+- reg: addresses and length of the register set for the device (must be at the same range as configured for
+  "marvell,armada-giu-nic" entry).
+  This is a set of two registers used by the management application to pass the base address of the configuration memory
+  space.
+  it also contains the msi registers space (which is used by the user-space to generate interrupts to host side).
+- reg-names: the namespace for each one of the mappings, specified in 'reg' property.
+
+Example:
+
+uio_agnic_0 {
+	compatible = "marvell,armada-giu-nic-uio";
+	reg = <0x6F0000 0x1000
+	       0x280000 0x1000>;
+	reg-names = "agnic_regs", "msi_regs";
+	dma-coherent;
+	status = "okay";
+};
diff --git a/Documentation/devicetree/bindings/net/marvell-armada-giu-nic.txt b/Documentation/devicetree/bindings/net/marvell-armada-giu-nic.txt
new file mode 100644
index 0000000..6faa6f6
--- /dev/null
+++ b/Documentation/devicetree/bindings/net/marvell-armada-giu-nic.txt
@@ -0,0 +1,20 @@
+* Marvell Armada GIU NIC (agnic)
+
+This entry is used by the loadable agnic driver (maintained out of the Linux tree)
+
+required properties:
+=========================
+- compatible: should be "marvell,armada-giu-nic"
+- reg: addresses and length of the register set for the device.
+  This is a set of two registers used by the management application to pass the base address of the configuration memory
+  space.
+
+Example:
+
+agnic-plat {
+	compatible = "marvell,armada-giu-nic";
+	reg = <0x6F00A0 0x8>;
+	msi-parent = <&gic_v2m0>;
+	dma-coherent;
+	status = "okay";
+};
diff --git a/arch/arm64/boot/dts/marvell/armada-ap806.dtsi b/arch/arm64/boot/dts/marvell/armada-ap806.dtsi
index 2b9fd37..1424dbe 100644
--- a/arch/arm64/boot/dts/marvell/armada-ap806.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-ap806.dtsi
@@ -519,6 +519,22 @@
 				compatible = "marvell,ap806-rev-info";
 				reg = <0x610fcc 0x4>;
 			};
+
+			agnic-plat {
+				compatible = "marvell,armada-giu-nic";
+				reg = <0x6F00A0 0x8>;
+				msi-parent = <&gic_v2m0>;
+				dma-coherent;
+				status = "okay";
+			};
+
+			uio_agnic_0 {
+				compatible = "marvell,armada-giu-nic-uio";
+				reg = <0x6F0000 0x1000
+				       0x280000 0x1000>;
+				reg-names = "agnic_regs", "msi_regs";
+				status = "okay";
+			};
 		};
 	};
 
diff --git a/arch/arm64/boot/dts/marvell/armada-ap810.dtsi b/arch/arm64/boot/dts/marvell/armada-ap810.dtsi
index c88ecb5..e0b10ad 100644
--- a/arch/arm64/boot/dts/marvell/armada-ap810.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-ap810.dtsi
@@ -201,6 +201,21 @@
 				status = "okay";
 				#thermal-sensor-cells = <0>;
 			};
+
+			AP810_LABEL(agnic): agnic-plat {
+				compatible = "marvell,armada-giu-nic";
+				reg = <0x6F00A0 0x8>;
+				/*msi-parent = <&AP810_LABEL(gic_its) 0x80>; no support for MSI for now */
+				dma-coherent;
+				status = "okay";
+			};
+			AP810_LABEL(uio_agnic): uio_agnic_0 {
+				compatible = "marvell,armada-giu-nic-uio";
+				reg = <0x6F0000 0x1000
+				       0x280000 0x1000>;
+				reg-names = "agnic_regs", "msi_regs";
+				status = "okay";
+			};
 		};
 	};
 };
-- 
2.7.4

